mmv1/third_party/terraform/services/storage/data_source_google_storage_buckets.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+package storage
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-google/google/tpgresource"
+	transport_tpg "github.com/hashicorp/terraform-provider-google/google/transport"
+)
+
+func DataSourceGoogleStorageBuckets() *schema.Resource {
+	return &schema.Resource{
+		Read: datasourceGoogleStorageBucketsRead,
+		Schema: map[string]*schema.Schema{
+			"prefix": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
At the moment the `project` input is required because using `tpgresource.GetProject(d, config)` didn't work during my tests.

The data source returned null when I tried to provide the project id via `GOOGLE_PROJECT` environment variable.

Reply:
I found the Issue.
It was a wrong usage of `d.SetId()` in line 113 before `return nil`

I changed `d.SetId(d.Get("project").(string))` to `d.SetId(params["project"])`

Now `project can be provided via provider or attribute